File: challenge/src/main/java/com/challenge/onboarding/product/model/ProductDetail.java
package com.challenge.onboarding.product.model;

import jakarta.persistence.*;

@Entity
@Table(name = "product_details")
public class ProductDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    private Double weight;

    @Column(columnDefinition = "json")
    private String dimensions;  // 예: {"width":10,"height":20,"depth":5}

    private String materials;

    @Column(name = "country_of_origin")
    private String countryOfOrigin;

    @Column(name = "warranty_info")
    private String warrantyInfo;

    @Column(name = "care_instructions", columnDefinition = "TEXT")
    private String careInstructions;

    @Column(name = "additional_info", columnDefinition = "jsonb")
    private String additionalInfo; // JSONB: 복잡한 키-값 저장

    protected ProductDetail() {}
}

File: src/product/domain/entities/Product_Detail.ts
export default class Product_Detail {
  constructor(
    public weight: number,
    public dimensions: {
      width: number;
      height: number;
      depth: number;
    },
    public materials: string,
    public country_of_origin: string,
    public warranty_info: string,
    public care_instructions: string,
    public additional_info: {
      assembly_required: boolean;
      assembly_time: string;
    },
    public product_id?: number,
    public id?: number,
  ) {}
}
